src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Feedback/RazorFeedbackDiagnosticFileProvider.cs

```diff
@@ -0,0 +1,112 @@
+            }
+
+            _feedbackLogDirectoryProvider = feedbackLogDirectoryProvider;
+            _joinableTaskFactory = joinableTaskContext.Factory;
+        }
+
+        public IReadOnlyCollection<string> GetFiles()
+        {
+            if (!_feedbackLogDirectoryProvider.DirectoryCreated)
+            {
+                // No one requested to create any feedback logs, no reason for us to provide any logs.
+                return Array.Empty<string>();
+            }
+
+            var timestamp = DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss", CultureInfo.InvariantCulture);
+            var zipFileName = $"RazorLogs_{timestamp}.zip";
+            var zipFilePath = Path.Combine(Path.GetTempPath(), zipFileName);
+            var logDirectory = _feedbackLogDirectoryProvider.GetDirectory();
+            if (!Directory.Exists(logDirectory))
+            {
+                // This should never be the case, just being extra defensive.
+                return Array.Empty<string>();
+            }
+
+            _joinableTaskFactory.RunAsync(() => ZipLogsAsync(logDirectory, zipFilePath));
+
+            return new[] { zipFilePath };
+        }
+
+        private static async Task ZipLogsAsync(string logs, string zipFilePath)
+        {
+            // Because we still have a handle to the log files, we can't zip them. So copy them and then zip the copy
+            var copiedLogs = CopyLogsToDirectory(logs);
+
+            // Start another task to zip this in the background.
+            await Task.Run(() => ZipFile.CreateFromDirectory(copiedLogs, zipFilePath));
+
+            try
+            {
+                if (Directory.Exists(copiedLogs))
+                {
+                    Directory.Delete(copiedLogs, true);
+                }
+            }
+            catch (Exception)
+            {
+                // Swallow any cleanup exceptions, we did our best.
+            }
+        }
+
+        private static string CopyLogsToDirectory(string logs)
+        {
+            var copyLocation = logs + DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss", CultureInfo.InvariantCulture);
+            if (!Directory.Exists(copyLocation))
+            {
+                Directory.CreateDirectory(copyLocation);
+                var dirInfo = new DirectoryInfo(logs);
+                foreach (var file in dirInfo.GetFiles())
+                {
+                    try
+                    {
+                        var toFilepath = Path.Combine(copyLocation, file.Name);
+                        file.CopyTo(toFilepath);
+                    }
+                    catch (Exception)
+                    {
+                        // If we can't copy a log file, indicate that there was one that failed to copy
+                        var failedFilePath = Path.Combine(copyLocation, "FAILED_" + file.Name);
+                        using var failedFile = File.Create(failedFilePath);
```

Comment:
Probably overkill, but what if this throws?

Reply:
lol, the world explodes 😄